### PR TITLE
Setup architecture for day selection and epilogue for blogging reminders

### LINKED ISSUE #14752: Blogging Reminders: Add logic to switch between screens
We have 3 screens - introduction, date selection and epilogue. We need to add logic to switch between these three screens.

### PR DESCRIPTION
**Pull request by planarvoid**
Fixes #14752 

This PR introduces logic to switch between the 3 supported screen for blogging reminders - prologue, day selection and epilogue. There is only the primary button visible now on the day selection and epilogue. Based on this PR it should be easily possible to build the UI part of the day selection and of the epilogue screen since the methods are ready. 

To test:
- Turn on the blogging reminders flag (Me/App settings/Test feature configuration - BloggingRemindersFeatureConfig)
- Create a new post
- Notice the blogging reminders prologue screen is shown
- Click on "Set goals"
- Notice the empty screen with disabled button is shown

To test: 
- Turn on the blogging reminders flag (Me/App settings/Test feature configuration - BloggingRemindersFeatureConfig)
- Go to site settings 
- Click on "Blogging goals"
- Notice the empty screen with disabled button is shown

## Regression Notes
1. Potential unintended areas of impact
None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by renanferrari — APPROVED** [review]
Good job on this! Code is looking good and I can confirm the app is working as expected.

LGTM 👍

Feel free to merge this once we merge the base branch.

File: WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/BloggingRemindersViewModel.kt
Comment (ravishanker) [inline WordPress/src/main/java/org/wordpress/android/ui/bloggingreminders/BloggingRemindersViewModel.kt:143]: Could we use [SavedStateHandle](https://developer.android.com/topic/libraries/architecture/viewmodel-savedstate) here!

**Review by ravishanker — APPROVED** [review]

**Pull request merged**
